src/cascadia/TerminalSettingsModel/Theme.cpp

```diff
@@ -0,0 +1,339 @@
+        }                                                                                      \
+                                                                                               \
+        Json::Value ToJson(const nameSpace::name& val)                                         \
+        {                                                                                      \
+            if (val == nullptr)                                                                \
+                return Json::Value::null;                                                      \
+            Json::Value json{ Json::ValueType::objectValue };                                  \
+            macro(THEME_SETTINGS_TO_JSON);                                                     \
+            return json;                                                                       \
+        }                                                                                      \
+                                                                                               \
+        std::string TypeDescription() const                                                    \
+        {                                                                                      \
+            return "name (You should never see this)";                                         \
+        }                                                                                      \
+    };
+
+THEME_OBJECT_CONVERTER(winrt::Microsoft::Terminal::Settings::Model, WindowTheme, MTSM_THEME_WINDOW_SETTINGS);
+THEME_OBJECT_CONVERTER(winrt::Microsoft::Terminal::Settings::Model, TabRowTheme, MTSM_THEME_TABROW_SETTINGS);
+
+#undef THEME_SETTINGS_FROM_JSON
+#undef THEME_SETTINGS_TO_JSON
+#undef THEME_OBJECT_CONVERTER
+
+Theme::Theme() noexcept :
+    Theme{ winrt::Windows::UI::Xaml::ElementTheme::Default }
+{
+}
+
+Theme::Theme(const winrt::Windows::UI::Xaml::ElementTheme& requestedTheme) noexcept
+{
+    auto window{ winrt::make_self<implementation::WindowTheme>() };
+    window->RequestedTheme(requestedTheme);
+    _Window = *window;
+}
+
+winrt::com_ptr<Theme> Theme::Copy() const
+{
+    auto theme{ winrt::make_self<Theme>() };
+
+    theme->_Name = _Name;
+
+    if (_Window)
+    {
+        theme->_Window = *winrt::get_self<implementation::WindowTheme>(_Window)->Copy();
+    }
+    if (_TabRow)
+    {
+        theme->_TabRow = *winrt::get_self<implementation::TabRowTheme>(_TabRow)->Copy();
+    }
+
+    return theme;
+}
+
+// Method Description:
+// - Create a new instance of this class from a serialized JsonObject.
+// Arguments:
+// - json: an object which should be a serialization of a ColorScheme object.
+// Return Value:
+// - Returns nullptr for invalid JSON.
+winrt::com_ptr<Theme> Theme::FromJson(const Json::Value& json)
+{
+    auto result = winrt::make_self<Theme>();
+    result->LayerJson(json);
+    return result;
+}
+
+void Theme::LayerJson(const Json::Value& json)

```

Comment:
If we can avoid it, I'd _personally_ prefer if we don't add new `LayerJson` functions. The existing ones were a bit of a headache at times IMO... I think in this case, we can just inline this code into `FromJson` and remove `LayerJson`.